pgrx/src/datum/geo.rs

```diff
@@ -7,9 +7,30 @@
 //LICENSE All rights reserved.
 //LICENSE
 //LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
-use crate::{pg_sys, FromDatum, IntoDatum, PgMemoryContexts};
 
-impl FromDatum for pg_sys::BOX {
+//! This module contains implementations to make it easier to work with Postgres' geometric types
+//! in Rust.
+//!
+//! Simple types that support `Copy` are re-exported from the `pg_sys` crate and have `IntoDatum`
+//! and `FromDatum` trait implementations. Variable Length types have zero-copy and owned structs
+//! that have `IntoDatum` and `FromDatum` trait implementations.
+//!
+//! See the [Postgres docs](https://www.postgresql.org/docs/current/datatype-geometric.html) for
+//! more information on the geometric types.
+use std::{mem, ptr};
+
+use pgrx_pg_sys::pfree;
+use pgrx_sql_entity_graph::metadata::{
+    ArgumentError, Returns, ReturnsError, SqlMapping, SqlTranslatable,
+};
+
+use crate::{pg_sys, set_varsize_4b, FromDatum, IntoDatum, PgMemoryContexts};
+
+// Copy types
+
+pub type Box = pg_sys::BOX;
```

Comment:
In [pgrx/src/datum/date.rs] the pattern is to do a transparent wrapping struct and add `From` impls. If that's the preferred pattern I can switch these over to that approach.

Reply:
Give us a few days to look this over.  

I don’t know what the right answer is. I think we might prefer a newtype but then that’ll cause complications over in pl/rust land for types that are already exposed, like BOX. That’s not your responsibility of course, buts it’s something we need to take into consideration.

Reply:
I've been thinking about this a bit more.  I think it's fine to leave these as aliases.  In fact, it's probably preferred.

A newtype wrapper would imply that pgrx has some more knowledge about the type and, for example, enforces bounds or requires some invariants that we don't.

We basically had to do that for the date/time types b/c even tho the underlying type is something basic like a u32 (or whatever), not all u32s are valid dates or timestamps or whatnot.  I don't guess that's the case here.

It looks like Postgres does some validation on the string representation of these types, of course, but it doesn't care to check the actual values.

Reply:
> We basically had to do that for the date/time types b/c even tho the underlying type is something basic like a u32 (or whatever), not all u32s are valid dates or timestamps or whatnot.

Yes, that is exactly the case. The types have implicit invariants that aren't going to be enforced by the "raw" `pg_sys` side, and lifting them into Rust allowed enforcing invariants. If **all pairs of points** are _valid BOXes_ (I'm talking only about Postgres's conception of validity), then it's fine to just use an alias.